pkg/controller/controller.go

```diff
@@ -106,11 +111,10 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) error {
 	var wg sync.WaitGroup
 	for i := 0; i < workers; i++ {
 		wg.Add(1)
-		// TODO (@munnerz): make time.Second duration configurable
 		go wait.Until(func() {
 			defer wg.Done()
 			c.worker(ctx)
-		}, time.Second, stopCh)
+		}, c.syncPeriod, stopCh)
```

Comment:
I've been squinting at this for a while and it seems to me that the use of `wait.Until` here is wrong.

`c.worker` is a blocking, long-running function which blocks while waiting for new items to be available on the queue and which only exits if the queue is shutting down.

So why would we want to ever want to re-run it after 1 second?

And calling `defer wg.Done()` inside the `Until` function also seems wrong because it would potentially mark as "done", other work group items.

Here's how controller-runtime does it:
 * https://github.com/kubernetes-sigs/controller-runtime/blob/e10bf72058d349e9c5cdd177593c32c650065458/pkg/internal/controller/controller.go#L206-L217

Reply:
In case it panics, if I recall correctly - this is taken from the sample-controller code

Reply:
sample-controller doesn't use a `sync.WaitGroup` here though and it has a separate panic handling line right at the start of the Run function:
https://github.com/kubernetes/sample-controller/blob/73e81dab82c945b087f7db1f1f4a7b50e7d92751/controller.go#L147-L175

It does call `wait.Until` with a 1 second interval, and as you say, that does include panic handling:

https://github.com/kubernetes/kubernetes/blob/e8cf412e5ec70081477ad6f126c7f7ef7449109c/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go#L136-L173

But the panic handling doesn't actually recover from panic, it calls some handlers and then continues to panic:

https://github.com/kubernetes/kubernetes/blob/e8cf412e5ec70081477ad6f126c7f7ef7449109c/staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go#L39-L58

So I'm still confused about this code.
But I am now sure that this particular time interval doesn't need to be configurable.

@davidsbond I propose that we close this PR